### DISCUSSION TITLE
how to show file in finder/explorer?

### DISCUSSION THREAD

**Discussion by thep0y** [Byron/open-rs#88] [Q&A] [answered]
Is there any API that does the same thing as the code below?

```rust
// Windows
Command::new("explorer")
        .args(["/select,", "\"path\\to\\file""]) // The comma after select is not a typo
        .spawn()
        .unwrap();
// Alternatively there is a win32 api for this iirc which needs to be used for longer file paths i think.

// macOS
Command::new( "open" )
        .args(["-R", "path/to/file"]) // i don't have a mac so not 100% sure
        .spawn()
        .unwrap();
```

**Comment by Byron** [comment, marked as answer]
You could see for yourself using `dbg!(open::commands("path/to/file")` ([API docs](https://docs.rs/open/latest/open/fn.commands.html)). For MacOS, the code is conveniently [just a few lines](https://github.com/Byron/open-rs/blob/main/src/macos.rs).

**Reply by thep0y** [reply]
I might not have expressed it clearly; what I mean is that `open` should not only directly open the file but also be able to display the file in the file manager. This should be achievable with just a small amount of code and would be quite useful in certain scenarios.

**Reply by Byron** [reply]
This means you'd want to open the containing directory, which should already work today (at least it does on MacOS).
On MacOS, using `-R` will also do that, but additionally highlight the file that was passed, which would be new.

With that said, the exact feature you are requesting isn't available even though the approximation (showing the containing directory) exists.

If you are interested in contributing this feature for all platforms and 'opener' programs, you are definitely welcome to create a PR for it.

**Reply by ngosang** [reply]
@Byron this will be very useful for Immich => github.com/qarmin/czkawka/issues/1522

**Comment by belst** [comment]
on windows you could also use [`SHOpenFolderAndSelectItems`](https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shopenfolderandselectitems)
